driver.go

```diff
@@ -126,14 +129,99 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 		maxap, err := mc.getSystemVar("max_allowed_packet")
 		if err != nil {
 			mc.Close()
-			return nil, err
+			return err
 		}
 		mc.maxAllowedPacket = stringToInt(maxap) - 1
 	}
 	if mc.maxAllowedPacket < maxPacketSize {
 		mc.maxWriteSize = mc.maxAllowedPacket
 	}
 
+	return err
+}
+
+//Connect opens a new connection without using a DSN
+func (c MySQLConnector) Connect(cxt context.Context) (driver.Conn, error) {
+	var err error
+
+	//Validate the connection parameters
+	//the following are required User,Pass,Net,Addr,DBName
+	//Pass may be blank
+	//The other optional parameters are not checks
+	//as GO will automatically enforce proper bool types on the options
+	if len(c.Cfg.User) > 32 || len(c.Cfg.User) <= 0 {
+		return nil, errInvalidUser
+	}
+
+	if len(c.Cfg.Addr) <= 0 {
+		return nil, errInvalidAddr
+	}
+
+	if len(c.Cfg.DBName) <= 0 {
+		return nil, errInvalidDBName
+	}
```

Comment:
Empty DB name should be supported.

Reply:
I concur. I had not thought of the SQL queries that do not require a database. I will correct.

Reply:
I use it when I create monitoring tools.  (e.g. `SHOW VARIABLES` or `SHOW FULL PROCESSLIST`)